__tests__/app.test.js

```diff
@@ -169,3 +169,102 @@ describe("Errors", () => {
+      .send(requestBody)
+      .expect(201)
+      .then((res) => {
+        const postedComment = res.body.comment;
+        expect(postedComment).toEqual(
+          expect.objectContaining({
+            comment_id: expect.any(Number),
+            body: expect.any(String),
+            article_id: expect.any(Number),
+            author: expect.any(String),
+            votes: expect.any(Number),
+            created_at: expect.any(String),
+          })
+        );
+      });
+  });
+
+  describe("Errors", () => {
+    describe("ID Errors", () => {
+      it("404: Should return a Not Found message when endpoint at the given ID does not exist", () => {
+        const requestBody = {
+          username: "butter_bridge",
+          body: "I buttered a butter bridge",
+        };
+        return request(app)
+          .post("/api/articles/99999/comments")
+          .send(requestBody)
+          .expect(404)
+          .then((res) => {
+            const body = res.body;
+            expect(body).toEqual({ message: "Not Found" });
+          });
+      });
+
+      it("400: Should return a Bad Request error when endpoint provided an ID that is the incorrect data type", () => {
+        return request(app)
+          .post("/api/articles/thisshouldcausea400errormessage/comments")
+          .expect(400)
+          .then((res) => {
+            const body = res.body;
+            expect(body).toEqual({ message: "Bad Request" });
+          });
+      });
+    });
+
+    it("400: Should respond with a Bad Request error message when sent a object with a malformed body", () => {
+      const requestBody = {};
+      return request(app)
+        .post("/api/articles/2/comments")
+        .send(requestBody)
+        .expect(400)
+        .then((res) => {
+          const body = res.body;
+          expect(body).toEqual({ message: "Bad Request" });
+        });
+    });
+
+    it("400: Should respond with an appropriate error message when the user sends a comment object with an incorrect data type", () => {
+      const requestBody = {
+        username: "butter_bridge",
+        body: { wad: 22 },
+      };
+      return request(app)
+        .post("/api/articles/2/comments")
+        .send(requestBody)
+        .expect(400)
+        .then((res) => {
+          const body = res.body;
+          expect(body).toEqual({ message: "Bad Request" });
+        });
+    });
+
+    describe("Username Errors", () => {
+      it("404: Should respond with a Not Found message when passed a username which is the correct data type but does not exist", () => {
+        const requestBody = {
+          username: "butter",
+          body: "string",
+        };
+        return request(app)
+          .post("/api/articles/2/comments")
+          .send(requestBody)
+          .expect(404)
+          .then((res) => {
+            const body = res.body;
+            expect(body).toEqual({ message: "Not Found" });
+          });
+      });
+    });
+  });
+});
```

Comment:
Really nice testing suite so far Tomasz 👍 Could you just add one more test for a 201 which would ignore unnecessary properties (i.e. if you add another property to the requestBody apart from username and body)